### PR TITLE
feat: new deposits table component

### PR DESCRIPTION
**Pull request by dohaki**
Requires https://github.com/across-protocol/frontend-v2/pull/907

Closes ACX-1623

Implements the updated deposits table. Currently only viewable via Storybook. Should therefore be tested by pulling the branch and running it locally.

For QA, it would make sense to involve Ryan and Tim in one of the upcoming PRs where I will integrate the component into the transactions page.

Some screenshots:

Full width
image

With overflow
image

Fee breakdown on hover
image

Transactions dropdown
image

Slow fill info tooltip
image

### CODE REVIEW

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { Text, Badge } from "components";
 
 import { ChainId, QUERIESV2, COLORS, NoFundsDepositedLogError } from "utils";
 
-import { useElapsedSeconds } from "../hooks/useElapsedSeconds";
+import { useElapsedSeconds } from "../../../hooks/useElapsedSeconds";
```
Comment (james-a-morris) [inline src/views/DepositStatus/components/DepositStatusUpperCard.tsx:19]: Could we reference from root here? `from "hooks/useElapsedSeconds"`

File: src/views/Bridge/hooks/useBridgeAction.ts [hunk 2]
```diff
@@ -107,6 +107,7 @@ export function useBridgeAction(
       originChainId: String(frozenPayload.fromChain),
       destinationChainId: String(frozenPayload.toChain),
       bridgeTokenSymbol: tokenSymbol,
+      referrer,
```
Comment (james-a-morris) [inline src/views/Bridge/hooks/useBridgeAction.ts:110]: Good call +1

File: src/hooks/useTokenConversion.ts
Comment (james-a-morris) [inline src/hooks/useTokenConversion.ts:1]: Was this copied from an open PR?

File: src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -50,10 +54,11 @@ export const Tooltip: React.FC<TooltipProps> = ({
       <StyledAnchor data-tooltip-id={id} data-tooltip-place={placement}>
         {children}
       </StyledAnchor>
-      <StyledTooltip id={id} noArrow opacity={1}>
-        <Wrapper>
+      <StyledTooltip id={id} noArrow opacity={1} offset={offset}>
+        <Wrapper maxWidth={maxWidth}>
           {title && (
             <TitleRow>
+              {typeof icon === "object" && icon}
```
Comment (james-a-morris) [inline src/components/Tooltip/Tooltip.tsx:61]: QQ: is this the case where we pass an arbitrary value? If so, would it make sense to resolve the icon prior to this rendering. I.e. `const Icon = (icon === "green-checkmark") ? RoundedCheckmark16 : (icon === "grey-checkmark")... : icon

File: src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -25,11 +25,13 @@ export type TooltipIcon =
   | "clock";
 export interface TooltipProps {
   tooltipId?: string;
-  icon?: TooltipIcon;
+  icon?: TooltipIcon | ReactNode;
```
Comment (james-a-morris) [inline src/components/Tooltip/Tooltip.tsx:28]: QQ: is there a difference between `ReactNode` vs `ReactElement`

File: src/components/DepositsTable/cells/RateCell.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+type Props = {
+  deposit: Deposit;
+  width: number;
+};
+
+export function RateCell({ deposit, width }: Props) {
+  return (
+    <StyledRateCell width={width}>
+      <Text color="light-200">
+        {deposit.rewards ? `${deposit.rewards.rate * 100}%` : "-"}
```
Comment (james-a-morris) [inline src/components/DepositsTable/cells/RateCell.tsx:17]: Should we floor/ceil or fix this decimal point?

File: src/components/DepositsTable/cells/DateCell.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import styled from "@emotion/styled";
+import { DateTime } from "luxon";
```
Comment (james-a-morris) [inline src/components/DepositsTable/cells/DateCell.tsx:2]: QQ: what is `luxon` and can we use the baked in Date class?

File: src/components/DepositsTable/HeadRow.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import styled from "@emotion/styled";
+
+import { Text } from "components/Text";
+
+import { COLORS } from "utils";
+
+export type HeaderCells = typeof headerCells;
+
+export type ColumnKey = keyof HeaderCells;
```
Comment (james-a-morris) [inline src/components/DepositsTable/HeadRow.tsx:9]: Nit: we should remove the white spaces to make this more cohesive. 
```suggestion
import styled from "@emotion/styled";
import { Text } from "components/Text";
import { COLORS } from "utils";

export type HeaderCells = typeof headerCells;
export type ColumnKey = keyof HeaderCells;
```

**Review by james-a-morris — COMMENTED** [review]
Left a few comments. I'd like to review again once QA is done

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 1 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx:19]: Oh yea, my IDE did some automatic things here. Will change

File: src/hooks/useTokenConversion.ts
Comment (dohaki) [reply src/hooks/useTokenConversion.ts:1]: Yea but I actually don't need it. Will remove

File: src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (dohaki) [reply src/components/Tooltip/Tooltip.tsx:28]: Yea, in 99% of cases, you should use `ReactNode`. The type allows everything React can render, i.e. string, number, undefined, null, JSX, etc. `ReactElement` aka `JSX.Element` on the other only allows JSX.

File: src/components/DepositsTable/HeadRow.tsx (hunk 7 above)
Comment (dohaki) [reply src/components/DepositsTable/HeadRow.tsx]: https://github.com/across-protocol/frontend-v2/pull/908/commits/fba48dc6b508defe324363d5550f12a896246bb6

File: src/components/Tooltip/Tooltip.tsx (hunk 3 above)
Comment (dohaki) [reply src/components/Tooltip/Tooltip.tsx]: https://github.com/across-protocol/frontend-v2/pull/908/commits/fba48dc6b508defe324363d5550f12a896246bb6

**Review by james-a-morris — APPROVED** [review]
LGTM pending QA

**Pull request merged**